core/block.js

```diff
@@ -1635,6 +1637,9 @@ Blockly.Block.prototype.appendInput_ = function(type, name) {
   if (type == Blockly.INPUT_VALUE || type == Blockly.NEXT_STATEMENT) {
     connection = this.makeConnection_(type);
   }
+  if (type == Blockly.NEXT_STATEMENT) {
```

Comment:
If you're doing this here, you need to also check again and possibly set it to false in `removeInput`

The other option is to figure out whether there are statement inputs in the renderer.  iirc the rows know it, but the top-level info object doesn't.

Reply:
Agree to using a counter instead of a boolean.
Renderer option won't work without a look-ahead as the info we need is in the top row before the other rows and inputs are created / added.